### ISSUE TITLE
Error when creating a store view if there are CMS pages with the same URL key

### ISSUE THREAD

**Issue by stupachenko** [magento/magento2#28357]
### Preconditions (*)

1. Magento 2.4-develop

### Steps to reproduce (*)

1. Navigate to Admin Panel > Stores > All Stores and create 2 store views:
```
Name: German
Code: german
Status: Enabled
```
```
Name: French
Code: french
Status: Enabled
```
2. Navigate to Admin Panel > Content > Pages and create 2 pages:
```
Page Title: About Us
URL Key: about-us
Store View: French
```
```
Page Title: About Us
URL Key: about-us
Store View: German
```
3. Navigate to Admin Panel > Stores > All Stores and create store view
```
Name: Spanish
Code: spanish
Status: Enabled
```

### Expected result (*)

1. Store view created successfully.

### Actual result (*)

1. Error message "Something went wrong while saving. Please review the error log."
```
[2020-05-25 10:46:30] main.CRITICAL: Exception message: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'about-us-4' for key 'URL_REWRITE_REQUEST_PATH_STORE_ID', query was: INSERT  INTO `url_rewrite` (`redirect_type`,`is_autogenerated`,`metadata`,`description`,`store_id`,`entity_type`,`entity_id`,`request_path`,`target_path`) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?)
Trace: <pre>#1 Magento\Framework\DB\Adapter\Pdo\Mysql->query() called at [generated/code/Magento/Framework/DB/Adapter/Pdo/Mysql/Interceptor.php:128]
#2 Magento\Framework\DB\Adapter\Pdo\Mysql\Interceptor->query() called at [lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php:2105]
#3 Magento\Framework\DB\Adapter\Pdo\Mysql->insertArray() called at [generated/code/Magento/Framework/DB/Adapter/Pdo/Mysql/Interceptor.php:557]
#4 Magento\Framework\DB\Adapter\Pdo\Mysql\Interceptor->insertArray() called at [lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php:2062]
#5 Magento\Framework\DB\Adapter\Pdo\Mysql->insertMultiple() called at [generated/code/Magento/Framework/DB/Adapter/Pdo/Mysql/Interceptor.php:544]
#6 Magento\Framework\DB\Adapter\Pdo\Mysql\Interceptor->insertMultiple() called at [app/code/Magento/UrlRewrite/Model/Storage/DbStorage.php:337]
#7 Magento\UrlRewrite\Model\Storage\DbStorage->insertMultiple() called at [app/code/Magento/UrlRewrite/Model/Storage/DbStorage.php:287]
#8 Magento\UrlRewrite\Model\Storage\DbStorage->doReplace() called at [app/code/Magento/UrlRewrite/Model/Storage/AbstractStorage.php:87]
#9 Magento\UrlRewrite\Model\Storage\AbstractStorage->replace() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:58]
#10 Magento\UrlRewrite\Model\Storage\DbStorage\Interceptor->___callParent() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:138]
#11 Magento\UrlRewrite\Model\Storage\DbStorage\Interceptor->Magento\Framework\Interception\{closure}() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:153]
#12 Magento\UrlRewrite\Model\Storage\DbStorage\Interceptor->___callPlugins() called at [generated/code/Magento/UrlRewrite/Model/Storage/DbStorage/Interceptor.php:65]
#13 Magento\UrlRewrite\Model\Storage\DbStorage\Interceptor->replace() called at [app/code/Magento/CmsUrlRewrite/Plugin/Cms/Model/Store/View.php:77]
#14 Magento\CmsUrlRewrite\Plugin\Cms\Model\Store\View->afterSave() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:146]
#15 Magento\Store\Model\ResourceModel\Store\Interceptor->Magento\Framework\Interception\{closure}() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:153]
#16 Magento\Store\Model\ResourceModel\Store\Interceptor->___callPlugins() called at [generated/code/Magento/Store/Model/ResourceModel/Store/Interceptor.php:117]
#17 Magento\Store\Model\ResourceModel\Store\Interceptor->save() called at [lib/internal/Magento/Framework/Model/AbstractModel.php:654]
#18 Magento\Framework\Model\AbstractModel->save() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:58]
#19 Magento\Store\Model\Store\Interceptor->___callParent() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:138]
#20 Magento\Store\Model\Store\Interceptor->Magento\Framework\Interception\{closure}() called at [app/code/Magento/Theme/Model/Indexer/Design/Config/Plugin/Store.php:39]
#21 Magento\Theme\Model\Indexer\Design\Config\Plugin\Store->aroundSave() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:135]
#22 Magento\Store\Model\Store\Interceptor->Magento\Framework\Interception\{closure}() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:153]
#23 Magento\Store\Model\Store\Interceptor->___callPlugins() called at [generated/code/Magento/Store/Model/Store/Interceptor.php:1040]
#24 Magento\Store\Model\Store\Interceptor->save() called at [app/code/Magento/Backend/Controller/Adminhtml/System/Store/Save.php:72]
#25 Magento\Backend\Controller\Adminhtml\System\Store\Save->processStoreSave() called at [app/code/Magento/Backend/Controller/Adminhtml/System/Store/Save.php:130]
#26 Magento\Backend\Controller\Adminhtml\System\Store\Save->execute() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:58]
#27 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->___callParent() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:138]
#28 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->Magento\Framework\Interception\{closure}() called at [lib/internal/Magento/Framework/App/Action/Plugin/ActionFlagNoDispatchPlugin.php:51]
#29 Magento\Framework\App\Action\Plugin\ActionFlagNoDispatchPlugin->aroundExecute() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:135]
#30 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->Magento\Framework\Interception\{closure}() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:153]
#31 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->___callPlugins() called at [generated/code/Magento/Backend/Controller/Adminhtml/System/Store/Save/Interceptor.php:26]
#32 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->execute() called at [lib/internal/Magento/Framework/App/Action/Action.php:110]
#33 Magento\Framework\App\Action\Action->dispatch() called at [app/code/Magento/Backend/App/AbstractAction.php:150]
#34 Magento\Backend\App\AbstractAction->dispatch() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:58]
#35 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->___callParent() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:138]
#36 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->Magento\Framework\Interception\{closure}() called at [app/code/Magento/Backend/App/Action/Plugin/Authentication.php:143]
#37 Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:135]
#38 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->Magento\Framework\Interception\{closure}() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:153]
#39 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->___callPlugins() called at [generated/code/Magento/Backend/Controller/Adminhtml/System/Store/Save/Interceptor.php:39]
#40 Magento\Backend\Controller\Adminhtml\System\Store\Save\Interceptor->dispatch() called at [lib/internal/Magento/Framework/App/FrontController.php:186]
#41 Magento\Framework\App\FrontController->processRequest() called at [lib/internal/Magento/Framework/App/FrontController.php:118]
#42 Magento\Framework\App\FrontController->dispatch() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:58]
#43 Magento\Framework\App\FrontController\Interceptor->___callParent() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:138]
#44 Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}() called at [lib/internal/Magento/Framework/Interception/Interceptor.php:153]
#45 Magento\Framework\App\FrontController\Interceptor->___callPlugins() called at [generated/code/Magento/Framework/App/FrontController/Interceptor.php:26]
#46 Magento\Framework\App\FrontController\Interceptor->dispatch() called at [lib/internal/Magento/Framework/App/Http.php:116]
#47 Magento\Framework\App\Http->launch() called at [generated/code/Magento/Framework/App/Http/Interceptor.php:24]
#48 Magento\Framework\App\Http\Interceptor->launch() called at [lib/internal/Magento/Framework/App/Bootstrap.php:260]
#49 Magento\Framework\App\Bootstrap->run() called at [pub/index.php:40]
</pre> [] []
```
### Additional Information
- steps and screens during confirmation https://github.com/magento/magento2/issues/28357#issuecomment-633863238

---

- [ ] Severity: **S2**

**Comment by swathishs** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @swathishs, here is your [Magento instance](https://i-28357-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-28357-2-4-develop.instances.magento-community.engineering/admin_461c](https://i-28357-2-4-develop.instances.magento-community.engineering/admin_461c)
Login: `919b85fe` Password: `80634986853c`
Instance will be terminated in up to 3 hours.

**Comment by swathishs** [comment]
The issue is been reproduced on the 2.4-develop branch
Please find the attachment below:
1. **Two store view:**
![image](https://user-images.githubusercontent.com/58288685/82873290-e61c4080-9f51-11ea-9879-b6d8af648ffc.png)

2. **Two Pages:**
![image](https://user-images.githubusercontent.com/58288685/82873402-149a1b80-9f52-11ea-8e51-9df3dc396117.png)

3. **Creating third store view (Error):**
![image](https://user-images.githubusercontent.com/58288685/82873478-32678080-9f52-11ea-90ae-5a72cb9679f2.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @swathishs 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-34637` were created

**Issue Available**: @swathishs, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @stupachenko. Thank you for your report. 
The issue has been fixed in magento/magento2#28421 by @engcom-Charlie in 2.4-develop branch
Related commit(s):
 - [035aaf08f8a5e5253a09bf5a7e618acf9d3e32f7](https://github.com/magento/magento2/commit/035aaf08f8a5e5253a09bf5a7e618acf9d3e32f7)
 - [f1d14503b3dd05743ce00eef444d7d248a78f6b6](https://github.com/magento/magento2/commit/f1d14503b3dd05743ce00eef444d7d248a78f6b6)
 - [9d417ec96f061e700316760c887679caab275e97](https://github.com/magento/magento2/commit/9d417ec96f061e700316760c887679caab275e97)

The fix will be available with the upcoming 2.4.1 release.
